.github/workflows/github.yml

```diff
@@ -138,41 +145,87 @@ jobs:
       uses: actions/checkout@v5
       with:
         submodules: recursive
+    
+    - name: Prepare APT staging dir
+      if: contains(matrix.os, 'ubuntu')
+      run: mkdir -p "$RUNNER_TEMP/apt-cache"
+    
+    - name: APT cache restore
+      if: contains(matrix.os, 'ubuntu')
+      id: aptcache
+      uses: actions/cache/restore@v4
+      with:
+        path: ${{ runner.temp }}/apt-cache
+        key: ${{ matrix.platform }}-apt-${{ matrix.os }}
+        restore-keys: |
+          ${{ matrix.platform }}-apt-${{ matrix.os }}
 
     - name: Prepare CI
       if: "${{ matrix.before_install != '' }}"
-      run: source '${{github.workspace}}/CI/before_install/${{matrix.before_install}}'
+      run: source '${{github.workspace}}/CI/before_install/${{matrix.before_install}}' '${{matrix.arch}}'
+    
+    # SAVE only if we didn't hit an existing cache
```

Comment:
Why? Shouldn't we refresh the cache?

Reply:
I quess we don't need to refresh components cache each run. Will always same, not like cache for code.